src/main.cpp

```diff
@@ -5269,12 +5274,13 @@ bool ProcessMessage(CNode* pfrom, std::string strCommand, CDataStream& vRecv, in
 
         if (pfrom->nVersion < MIN_PEER_PROTO_VERSION)
         {
-            // disconnect from peers older than this proto version
-            LogPrintf("peer=%d using obsolete version %i; disconnecting\n", pfrom->id, pfrom->nVersion);
+            // ban peers older than this proto version
             pfrom->PushMessage(NetMsgType::REJECT, strCommand, REJECT_OBSOLETE,
-                               strprintf("Version must be %d or greater", MIN_PEER_PROTO_VERSION));
-            pfrom->fDisconnect = true;
-            return false;
+                               strprintf("Protocol Version must be %d or greater", MIN_PEER_PROTO_VERSION));
+            LOCK(cs_main);
+            Misbehaving(pfrom->GetId(), 100);
```

Comment:
the intention of the old code is to allow a client to reconnect with the proper version -- imagine in the future there's an incompatible protocol change and so clients can connect to other clients in 2 "modes".

A client could repeatedly connect by issuing "connect, good version, disconnect".  This algorithm is equivalent to the "connect, bad version, forced disconnect" that we have today.  Therefore I'd suggest that we revert this change.  In the future, "misbehaving 1" for every rapid disconnect makes sense, especially if we allow too many rapid disconnects to result in just a few hour ban, and push that ban into the kernel network level via iptables.